packages/autocomplete-core/src/getPropGetters.ts

```diff
@@ -162,6 +168,11 @@ export function getPropGetters<
     const { inputElement, maxLength = 512, ...rest } = providedProps || {};
     const activeItem = getActiveItem(store.getState());
 
+    const enterKeyHint = activeItem?.itemUrl ? 'go' : 'search';
+    const userAgent = props.environment.navigator?.userAgent;
+    const enterKeyHintFallback =
+      isSamsung(userAgent) && isAndroid(userAgent) && isChrome(userAgent);
```

Comment:
How about these lines? (and no need for the one below)

```suggestion
    const userAgent = props.environment.navigator?.userAgent;
    const shouldFallbackKeyHint =
      isSamsung(userAgent) && isAndroid(userAgent) && isChrome(userAgent);
    const enterKeyHint = activeItem?.itemUrl && !shouldFallbackKeyHint ? 'go' : 'search';
```